allauth_2fa/forms.py

```diff
@@ -62,9 +65,13 @@ def __init__(self, user, **kwargs):
 
     def save(self):
         # Delete any backup tokens.
-        static_device = self.user.staticdevice_set.get(name='backup')
-        static_device.token_set.all().delete()
-        static_device.delete()
+        try:
+            static_device = self.user.staticdevice_set.get(name=app_settings.BACKUP_DEVICE_NAME)
```

Comment:
@akx I'm a bit confused by this change, how could this device *not* exist when it is being asked to be removed?

Reply:
Before this commit, simply visiting the `TwoFactorBackupTokens` caused the device to exist (`get_or_create()` in GET).

Now it takes actually POSTing into `TwoFactorBackupTokens` to make it exist; if the user has skipped creating backup tokens, this would crash.

Reply:
I understand that for the changes to `TwoFactorBackupTokens`, but this form is used in `TwoFactorRemove`.

Reply:
Yup. So now that we don't _always_ have a static device, the unguarded 

```python
static_device = self.user.staticdevice_set.get(name='backup')
```
here in the removal view would have crashed (`ObjectDoesNotExist`), and no static device nor any TOTP device would have gotten removed.

So with the static device not always existing, we need to guard against that case here.